discovery_engine_core/core/src/engine.rs

```diff
@@ -294,6 +294,10 @@ impl Engine {
         let search_config = SearchConfig {
             max_docs_per_batch: config.max_docs_per_search_batch as usize,
         };
+        #[cfg(feature = "storage")]
+        let data_dir = std::path::PathBuf::from(&config.data_dir);
+        #[cfg(feature = "storage")]
+        let use_in_memory_db = config.use_in_memory_db;
```

Comment:
Config is consumed below.

Generally I think there is a lot of potential to cleanup code around `new`, `from_config` the structure of the context etc. but this is outside of the scope of this PR.

Reply:
i agree, it's also better to finish the db migration first and after the dead code is removed, we can then check what's left to simplify.